torch/_inductor/sizevars.py

```diff
@@ -649,3 +649,6 @@ def store_reduction(self, name, index, value):
 
     def index_expr(self, index, dtype):
         return self._inner.index_expr(self._simplify(index), dtype)
+
+    def lazy_check_bounds(self, index, size, lower, upper):
+        return self._inner.check_bounds(self._simplify(index), size, lower, upper)
```

Comment:
Why did the name change here? I also don't see why we need to call this `lazy_check_bounds`, it seems to me the laziness is an artifact of loads being reordered in codegen more than an integral part of the semantics.

Reply:
You are totally right. This laziness artifact is a rather clean way to solve that problem.